src/domain/session/room/RoomBeingCreatedViewModel.ts

```diff
@@ -45,29 +58,29 @@ export class RoomBeingCreatedViewModel extends ViewModel {
         }
         return "";
     }
-    get avatarLetter() { return avatarInitials(this.name); }
-    get avatarColorNumber() { return getIdentifierColorNumber(this._roomBeingCreated.avatarColorId); }
-    get avatarTitle() { return this.name; }
+    get avatarLetter(): string { return avatarInitials(this.name); }
+    get avatarColorNumber(): number { return getIdentifierColorNumber(this._roomBeingCreated.avatarColorId); }
+    get avatarTitle(): string { return this.name; }
 
-    avatarUrl(size) {
+    avatarUrl(size: number): string | null {
         // allow blob url which doesn't need mxc => http resolution
         return this._roomBeingCreated.avatarBlobUrl ??
-            getAvatarHttpUrl(this._roomBeingCreated.avatarUrl, size, this.platform, this._mediaRepository);
+            getAvatarHttpUrl(this._roomBeingCreated.avatarUrl!, size, this.platform, this._mediaRepository);
     }
 
-    focus() {}
+    focus(): void {}
 
-    _onRoomChange() {
-        this.emitChange();
+    _onRoomChange(): void {
+        this.emitChange("TODO");
```

Comment:
@MidhunSureshR need some guidance here for what string this should emit

Reply:
You can use:
```suggestion
        this.emitChange("this._roomBeingCreated");
```